libraries/adaptive-expressions/src/functionUtils.ts

```diff
@@ -768,14 +770,43 @@ export class FunctionUtils {
             return obj1 == null && obj2 == null;
         }
 
-        if (Array.isArray(obj1) && obj1.length === 0 && Array.isArray(obj2) && obj2.length === 0) {
-            return true;
+        // Array Comparison
+        if (Array.isArray(obj1) && Array.isArray(obj2)) {
+            if (obj1.length !== obj2.length) {
+                return false;
+            }
+
+            let isEqual = true;
+            for (let i = 0; i < obj1.length; i++) {
```

Comment:
Could simplify this (incorporating early return) by writing
```typescript
return obj1.every((item, i) => FunctionUtils.commonEquals(item, obj2[i]));
```

Reply:
Good suggestion!